### PR TITLE
rest: simplifications on uri and add telemetry

### PR DESCRIPTION
**Pull request by yugangw-msft**
Add a few optimizations towards the e2e experience by starting with referencing ARM rest API doc:
1. A full uri is not required. If host is missing, assume ARM endpoint
2. Replace the  `{subscriptionId}` token with real id

Also add telemetry data.

//CC: @achandmsft  
Related issue: #7618

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -1769,10 +1769,10 @@ def list_resource_links(cmd, scope=None, filter_string=None):
 # endregion
 
 
-def rest_call(cmd, method, url, headers=None, uri_parameters=None,
+def rest_call(cmd, method, uri, headers=None, uri_parameters=None,
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: I would recommend making the `az rest` command `is_preview=True` until the signature stabilizes. One thing

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py [hunk 2]
```diff
@@ -1200,8 +1200,8 @@
     examples:
        - name: Get Audit log through Microsoft Graph
          text: >
-             az rest --method get --url https://graph.microsoft.com/beta/auditLogs/directoryAudits
-       - name: Update a Azure Active Directory Graph User's display name
+             az rest --method get --uri https://graph.microsoft.com/beta/auditLogs/directoryAudits
+       - name: Update a Azure Active Directory Graph User's display name in BASH
          text: >
-             az rest --method patch --headers "{\"Content-Type\": \"application/json\"}" --url "https://graph.microsoft.com/v1.0/users/johndoe@azuresdkteam.onmicrosoft.com" --body "{\"displayName\":\"jondoe2\"}"
+             az rest --method patch --uri "https://graph.microsoft.com/v1.0/users/johndoe@azuresdkteam.onmicrosoft.com" --body '{"displayName": "jondoe2"}'
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: This will not work on CMD.

Comment (adewaleo) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: @tjprescott because of the single quotes?

Comment (tjprescott) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: Yes.

**Pull request merged**